### PR TITLE
More unit tests

### PR DESCRIPTION
**Pull request by Joerg-Seitz**
pls review & merge

dunno what to write otherwise... it should all work...

### CODE REVIEW

**Comment by Joerg-Seitz** [comment]
is there a way to resolve the one pending rubocop issue (extra line) without fixing it locally and commit + push? with the other issues i could just edit the code and commit it with a button..

File: app/models/geometry/edge.rb [hunk 1]
```diff
@@ -6,24 +6,24 @@ def initialize(from, to)
...
-     (midpoint.last + (factor / 2 * vector.last))]
-  end
-
   def length
-    x_coordinate = (to.first - from.first)
-    y_coordinate = (to.last - from.last)
-    Math.sqrt(x_coordinate**2 + y_coordinate**2)
+    x_distance = (to.first - from.first)
+    y_distance = (to.last - from.last)
+    Math.sqrt(x_distance**2 + y_distance**2)
```
Comment (andreasknoepfle) [inline app/models/geometry/edge.rb]: 👍

File: spec/models/geometry/edge_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+describe Geometry::Edge do
+  let(:from) { [0, 0] }
+  let(:to) { [10, 10] }
+  subject { Geometry::Edge.new(from, to) }
+
+  it 'should calculate length' do
+    expect(subject.length).to eq 14.142135623730951
+  end
+
+  context 'with vector'
```
Comment (andreasknoepfle) [inline spec/models/geometry/edge_spec.rb]: this context does not open a block ... 

should probably be 
```
describe '#offset_point_by_half_of_length' do 
end
```

File: spec/models/geometry/edge_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  let(:vector) { [10, -10] }
+
+  it 'should offset the midpoint' do
+    expect(subject.offset_point_by_half_of_length(0)).to eq [5, 5]
+    expect(subject.offset_point_by_half_of_length(0.5)).not_to eq [5, 5]
+  end
```
Comment (andreasknoepfle) [inline spec/models/geometry/edge_spec.rb]: `.not_to eq [5, 5] `
rather than that you should add actual values
`.to eq [???, ???]`

Comment (andreasknoepfle) [inline spec/models/geometry/edge_spec.rb]: I'd also like this to be in different contexts, while one tests with the default param and one with a custom factor:
```
context 'with a factor of 0.5' do 
   let(:factor) { 0.5 }
   ... 
end

context 'with the default factor' do
   ... 
end
```

Comment (Joerg-Seitz) [inline spec/models/geometry/edge_spec.rb]: All done

File: spec/models/geometry/edge_spec.rb [hunk 4]
```diff
@@ -8,11 +8,19 @@
...
+      it 'should offset the midpoint' do
+        expect(subject.offset_point_by_half_of_length(0.5)).to eq [7.5, 2.5]
+      end
+    end
+
+    context 'with a factor of  0,5' do
+      it 'should offset the midpoint' do
+        expect(subject.offset_point_by_half_of_length).to eq [5, 5]
+      end
+    end
```
Comment (andreasknoepfle) [inline spec/models/geometry/edge_spec.rb]: sadly the test description is now not matching the actual test .... where there is 0.5 the description says default and the other way around :(

Comment (Joerg-Seitz) [inline spec/models/geometry/edge_spec.rb]: x( fixed

**Pull request merged**